lib/http.go

```diff
@@ -0,0 +1,71 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const jsonMimeType = "application/json"
+
+// HTTPClient implements the Scheduler interface over HTTP, talking to a
+// Cron HTTPServer
+type HTTPClient struct {
+	Addr string
+}
+
+func NewHTTPClient(addr string) *HTTPClient {
+	return &HTTPClient{
+		Addr: addr,
+	}
+}
+
+func (c HTTPClient) Call(action string, ctx context.Context, params interface{}, result interface{}) error {
+	if action == "TransformMethods.Apply" {
```

Comment:
Totally workable for a code spike, just wanted to point to how go rpc works. It has a function "suitableMethods" (see https://golang.org/src/net/rpc/server.go) that uses reflection to find all methods on an interface that matches a certain pattern, and then automatically exposes them over rpc. I hope we can do something similar for the http case, though we would need some way to map the method names to their respective endpoints.

Reply:
I firmly agree with this point of automatic/programmatic mapping the API layer from lib. I thought it'd be good to start with the apply command for this reason: the `/apply` endpoint is ripe for formalizing. So far it's just a JSON serialization wrapper to a lib method over HTTP.

Need to put more thought into how such a generalization would work, and we'll most certainly have endpoints that won't fit neatly into some "make an HTTP endpoint for this lib method" box, but I think we should still move in that direction.